File: codegen/results/v002/lmstudio/qwen_qwen3-coder-30b_nothink/20260827_225600/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	// Check if command line argument is provided
	if len(os.Args) < 2 {
		return
	}

	// Parse the command line argument
	N, err := strconv.Atoi(os.Args[1])
	if err != nil {
		return
	}

	// If N < 1, print nothing
	if N < 1 {
		return
	}

	// Generate and print Fibonacci numbers
	a, b := 1, 1
	
	// Print first Fibonacci number
	if a <= N {
		fmt.Println(a)
	}
	
	// Print second Fibonacci number
	if b <= N {
		fmt.Println(b)
	}
	
	// Generate and print remaining Fibonacci numbers
	for {
		next := a + b
		if next > N {
			break
		}
		fmt.Println(next)
		a, b = b, next
	}
}

File: codegen/results/v002/lmstudio/qwen_qwen3-coder-30b_nothink/20260827_225600/python/003_fibonacci/solution.py
import sys

def main():
    # Read the command line argument
    if len(sys.argv) != 2:
        print("Usage: python script.py N")
        return
    
    try:
        N = int(sys.argv[1])
    except ValueError:
        print("Please provide a valid integer")
        return
    
    # If N < 1, print nothing
    if N < 1:
        return
    
    # Generate Fibonacci numbers up to N
    a, b = 1, 1
    
    # Print the first Fibonacci number
    if a <= N:
        print(a)
    
    # Print the second Fibonacci number
    if b <= N:
        print(b)
    
    # Generate and print subsequent Fibonacci numbers
    while True:
        next_fib = a + b
        if next_fib > N:
            break
        print(next_fib)
        a, b = b, next_fib

if __name__ == "__main__":
    main()